eudemo/eudemo/maintenance/lower_port/designer.py

```diff
@@ -0,0 +1,595 @@
+        # divertor to fit through
+        div_width_at_inner = div_duct_inner_point[0] * np.tan(
+            np.deg2rad((360 / self.n_TF) / self.n_div_cassettes)
+        )
+        if (inner_y_point - self.lower_duct_wall_tk) < div_width_at_inner / 2:
+            raise GeometryError(
+                "LowerPortDesigner: duct wall thickness is too large for the"
+                f" space between TF coils at x={div_duct_inner_point[0]},"
+                " which would intersect the divertor at the inner point."
+                " Making the duct angle shallower or reducing the"
+                " duct wall thickness would help."
+            )
+
+        return (
+            make_polygon(
+                {
+                    "x": np.array(
+                        [
+                            wall_outer_point[0],
+                            wall_outer_point[0],
+                            wall_inner_point[0],
+                            wall_inner_point[0],
+                        ]
+                    ),
+                    "y": np.array(
+                        [
+                            outer_y_point,
+                            -outer_y_point,
+                            -inner_y_point,
+                            inner_y_point,
+                        ]
+                    ),
+                    "z": np.array(
+                        [
+                            wall_outer_point[2],
+                            wall_outer_point[2],
+                            wall_inner_point[2],
+                            wall_inner_point[2],
+                        ]
+                    ),
+                },
+                closed=True,
+            ),
+            outer_y_point,
+        )
+
+    def _duct_xz_shapes(
+        self,
+        top_outer_point: Tuple,
+        bottom_inner_point: Tuple,
+    ):
+        angled_duct_boundary = self._angled_duct_boundary(
+            top_outer_point,
+            bottom_inner_point,
+        )
+        angled_duct = BluemiraFace(
+            angled_duct_boundary,
+        )
+
+        straight_duct_z_top = (
+            self.tf_coil_xz_boundary.bounding_box.z_min
+            - self.lower_port_tf_z_offset
+            # the wall thickness is adds to the z, so subtract it here
+            # to make the actual top of the duct at the correct z
+            # (i.e. self.tf_coil_xz_boundary.bounding_box.z_min
+            # - self.lower_duct_tf_z_offset)
+            - self.lower_duct_wall_tk
+        )
+
+        straight_duct_boundary, bottom_left_point = self._straight_duct_boundary(
+            angled_duct_boundary,
+            z_top_point=straight_duct_z_top,
+        )
+        straight_duct = BluemiraFace(straight_duct_boundary)
+
+        angled_cuts = boolean_cut(
+            angled_duct,
+            [straight_duct],
+        )
+        top_piece = angled_cuts[0]
+        duct_inner = boolean_fuse([top_piece, straight_duct])
```

Comment:
Possibly overkill but a bit of rearrangment would be good as angled pieces is not really used
```suggestion
        duct_inner = boolean_fuse([boolean_cut(angled_duct, straight_duct)[0], straight_duct])
```

Reply:
Yea, thought it was more clear the way it is, but will change

Reply:
you could just get rid of the angled_cut variable as a half way house